src/deserialize.rs

```diff
@@ -0,0 +1,201 @@
+            _marker: PhantomData::default(),
+        }
+    }
+}
+
+impl<'de, T> Visitor<'de> for WktVisitor<T>
+where
+    T: FromStr + Default + Debug + WktFloat,
+{
+    type Value = Wkt<T>;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(formatter, "a valid WKT format")
+    }
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        Wkt::from_str(s).map_err(|e| serde::de::Error::custom(e))
+    }
+}
+
+impl<'de, T> serde::Deserialize<'de> for Wkt<T>
+where
+    T: FromStr + Default + Debug + WktFloat,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(WktVisitor::default())
+    }
+}
+
+struct GeometryVisitor<T> {
+    _marker: PhantomData<T>,
+}
+
+impl<T> Default for GeometryVisitor<T> {
+    fn default() -> Self {
+        GeometryVisitor {
+            _marker: PhantomData::default(),
+        }
+    }
+}
+
+impl<'de, T> Visitor<'de> for GeometryVisitor<T>
+where
+    T: FromStr + Default + WktFloat,
+{
+    type Value = Geometry<T>;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(formatter, "a valid WKT format")
+    }
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        let mut wkt = Wkt::from_str(s).map_err(|e| serde::de::Error::custom(e))?;
+        if wkt.items.len() == 1 {
+            Ok(wkt.items.remove(0))
+        } else {
+            Err(serde::de::Error::custom(
+                "WKT should have only 1 Geometry item",
+            ))
+        }
+    }
+}
+
+impl<'de, T> serde::Deserialize<'de> for Geometry<T>
+where
+    T: FromStr + Default + WktFloat,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(GeometryVisitor::default())
+    }
+}
+
+/// This is a helper function to convert directly from WKT format into a geo_types::Geometry.
+/// ```
+/// # extern crate wkt;
+/// # extern crate geo_types;
+/// # extern crate serde_json;
+/// use geo_types::Geometry;
+///
+/// #[derive(serde::Deserialize)]
+/// struct MyType {
+///     #[serde(deserialize_with = "wkt::deserialize::deserialize_wkt_geometry")]
+///     pub geometry: Geometry<f64>,
+/// }
+///
+/// let json = r#"{ "geometry": "POINT (3.14 42)" }"#;
+/// let my_type: MyType = serde_json::from_str(json).unwrap();
+/// assert!(matches!(my_type.geometry, Geometry::Point(_)));
+/// ```
+#[cfg(feature = "geo-types")]
+pub fn deserialize_wkt_geometry<'de, D>(
+    deserializer: D,
+) -> Result<geo_types::Geometry<f64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    use serde::Deserialize;
+    use std::convert::TryInto;
+    let geometry: Geometry<f64> = Geometry::deserialize(deserializer)?;
+    let geometry: geo_types::Geometry<f64> = geometry.try_into().unwrap();
```

Comment:
can't we export an error here?

*very* minor: wouldn't it be better to `use std::convert::TryInto;` just before using it?

Reply:
I agree we don't want to `unwrap` here, thanks for noticing @antoine-de

Reply:
I completely forgot that `.unwrap()`, oops. Fixed in d5a8de371219c84d9fff67a8e501d44f5c397a1d. I also reorganized the code a bit with a `.and_then()`, but please tell if you prefer the previous version, I can definitely change it back.

Reply:
I also made the helper function generic over the type `T` of `Geometry<T>` because the first version was useful for `Geometry<f64>` but was useless for `Geometry<f32>`. See commit be202f413bfc1f2dbdfff3d0333cb8c0606025f0.

Reply:
These changes all look good to me, thank you!